index.bs

```diff
@@ -5071,87 +5071,104 @@ No [=extended attributes=] are applicable to dictionaries.
+
+    <blockquote>
+        <p>[=exception/Throw=] a {{TypeError}}.
+    </blockquote>
 
-Note: See [[#es-creating-throwing-exceptions]] for details on what creating and throwing an exception
-entails in the ECMAScript language binding.
+    <p>To throw a new {{DOMException}} with [=error name=] "{{NotAllowedError}}":
 
-<div class="example" id="example-56ad390c">
+    <blockquote>
+        <p>[=exception/Throw=] an "{{NotAllowedError}}" {{DOMException}}.
+    </blockquote>
 
-    Here is are some examples of wording to use to create and throw exceptions.
-    To throw a new [=simple exception=] named {{TypeError}}:
+    <p>To create a new {{DOMException}} with [=error name=] "{{SyntaxError}}":
 
     <blockquote>
-        [=exception/Throw=] a {{TypeError}}.
+        <p>Let <var ignore>object</var> be a newly [=created=] "{{SyntaxError}}" {{DOMException}}.
     </blockquote>
 
-    To throw a new {{DOMException}} with [=error name=] "{{NotAllowedError!!exception}}":
+    <p>To [=reject=] a promise with a new {{DOMException}} with [=error name=] "{{OperationError}}":
 
     <blockquote>
-        [=exception/Throw=] an "{{NotAllowedError!!exception}}" {{DOMException}}.
+        [=Reject=] <var ignore>p</var> with an "{{OperationError}}" {{DOMException}}.
     </blockquote>
+</div>
 
-    To create a new {{DOMException}} with [=error name=] "{{SyntaxError!!exception}}":
+<div class=example id=example-exception-throwing-message-hint>
+    <p>An example of including additional information used to construct the [=exception/message=]
+    would be:
 
     <blockquote>
-        Let <var ignore>object</var> be a newly [=created=] "{{SyntaxError!!exception}}" {{DOMException}}.
+        [=exception/Throw=] a "{{SyntaxError}}" {{DOMException}} indicating that the given value
+        had disallowed trailing spaces.
     </blockquote>
+
+    <p>Such additional context is most helpful to implementers when it is not immediately obvious
+    why the exception is being thrown, e.g., because there are many different steps in the algorithm
+    which throw a "{{SyntaxError}}" {{DOMException}}. In contrast, if your specification throws a
+    "{{NotAllowedError}}" {{DOMException}} immediately after checking if the user has provided
+    permission to use a given feature, it's fairly obvious what sort of [=exception/message=] the
+    implementation should construct, and so specifying it is not necessary.
 </div>
 
+The resulting behavior from creating and throwing an exception is language binding-specific.
+
+<p class=note>See [[#es-creating-throwing-exceptions]] for details on what creating and throwing an
+exception entails in the ECMAScript language binding.
 
-<h4 id="idl-DOMException-error-names">Error names</h4>
+
+<h4 id="idl-DOMException-error-names">{{DOMException}} error names</h4>
 
 The <dfn id="dfn-error-names-table" export>error names table</dfn> below lists all the allowed error names
```

Comment:
Shouldn't this be rephrased, since other specs will now be able to introduce new ones?